src/garage/tf/policies/task_embedding_policy.py

```diff
@@ -0,0 +1,332 @@
+
+        Returns:
+            np.ndarray: Actions sampled from the policy.
+
+        """
+        raise NotImplementedError
+
+    def dist_info_sym(self, input_var, state_info_vars=None, name='default'):
+        """Symbolic graph of action distribution.
+
+        Return the symbolic distribution information about the actions.
+
+        This function is not implemented because Task Embedding policy requires
+        an additional task id to sample action.
+
+        Args:
+            input_var (tf.Tensor): symbolic variable for observations
+            state_info_vars (dict): a dictionary whose values should contain
+                information about the state of the policy at the time it
+                received the observation.
+            name (str): Name of the symbolic graph.
+
+        """
+        raise NotImplementedError
+
+    def dist_info(self, input_val, state_infos):
+        """Action distribution info.
+
+        Return the distribution information about the actions.
+
+        This function is not implemented because Task Embedding policy requires
+        an additional task id to sample action.
+
+        Args:
+            input_val (tf.Tensor): observation values
+            state_infos (dict): a dictionary whose values should contain
+                information about the state of the policy at the time it
+                received the observation
+
+        """
+        raise NotImplementedError
+
+    def get_trainable_vars(self):
+        """Get trainable variables.
+
+        The trainable vars of a multitask policy should be the trainable vars
+        of its model and the trainable vars of its embedding model.
+
+        Returns:
+            List[tf.Variable]: A list of trainable variables in the current
+                variable scope.
+
+        """
+        return (self._variable_scope.trainable_variables() +
+                self.encoder.get_trainable_vars())
+
+    def get_global_vars(self):
+        """Get global variables.
+
+        The global vars of a multitask policy should be the global vars
+        of its model and the trainable vars of its embedding model.
+
+        Returns:
+            List[tf.Variable]: A list of global variables in the current
+                variable scope.
+
+        """
+        return (self._variable_scope.global_variables() +
+                self.encoder.get_global_vars())
+
+    def split_task_observation(self, collated):
+        """Splits up observation into one-hot task and environment observation.
+
+        Args:
+            collated (np.ndarray): Environment observation concatenated with
+                task one-hot.
+
+        Returns:
+            np.ndarray: Vanilla environment observation.
+            np.ndarray: Task one-hot.
+
+        """
+        task_dim = self.task_space.flat_dim
+        return collated[:-task_dim], collated[-task_dim:]
+
+    @staticmethod
+    def concat_spaces(first, second):
```

Comment:
this feels like it should either be an akro feature or a helper method, but not hanging out in this class

Reply:
I moved it to akro. https://github.com/rlworkgroup/akro/pull/59